scripts/zones/Sacrarium/mobs/Old_Professor_Mariselle.lua

```diff
@@ -2,20 +2,30 @@
 -- Area: Sacrarium
 --  Mob: Old Professor Mariselle
 -----------------------------------
-local ID = require("scripts/zones/Sacrarium/IDs")
-require("scripts/globals/keyitems")
-require("scripts/globals/missions")
+local ID = require('scripts/zones/Sacrarium/IDs')
+local professorTables = require('scripts/zones/Sacrarium/globals')
+require('scripts/globals/keyitems')
+require('scripts/globals/missions')
+require('scripts/globals/utils')
 -----------------------------------
 local entity = {}
 
+entity.onMobSpawn = function(mob)
+    mob:setCarefulPathing(true)
+
+    for i = 0, 5 do
+        if GetNPCByID(ID.npc.QM_MARISELLE_OFFSET + i):getLocalVar('hasProfessorMariselle') == 1 then
+            mob:setLocalVar('spawnLocation', i)
+        end
+    end
+end
+
 entity.onMobFight = function(mob, target)
     local opMariselle = mob:getID()
 
-    -- Summons a pupil every 30 seconds.
-    -- TODO: Casting animation for summons. When he spawns them isn't retail accurate.
-    -- TODO: Make him and the clones teleport around the room every 30s
-
-    if (mob:getBattleTime() % 30 < 3 and mob:getBattleTime() > 3) then
+    if mob:getBattleTime() % 30 < 3 and
```

Comment:
didn't look that wide on github but if a wide conditional needs broken up you'll want to newline at the if also like:
```lua
if
    mob:getBattleTime() % 30 < 3 and
    mob:getBattleTime() > 3
then
```